cmd/samToBed/testdata/test2.bed

```diff
@@ -1,2 +1,2 @@
-ref	28	45	r003
-ref	36	45	r001
+ref	3	33	r003
```

Comment:
I am not sure why the samToBed test files are changing.  I don't think messing up the seq being from the reverse strand should mess with the coordinates.  I also quickly looked at test.sam, and it seems like 28 and 36 should be the start coordinates in test2.bed.

Reply:
I wonder if the changes have something to do with fragLength?
Since test1 didn't change in samToBed or samToWig, and only test2 changed,
and looking in the _test.go files fragLength is set to 30 (not the default -1) only in test2 not test1.
fragLength matters in a function in gonomics/convert/convert.go: SamToBedFrag (as opposed to SamToBed).
The way SamToBedFrag works does depend on positive/negative strand.

cmd/samToBed/testdata/test2.sam
@SQ	SN:ref	LN:45
r003	16	ref	29	30	6H5M	*	0	0	TAGGC	*
r001	83	ref	37	30	9M	=	7	-39	CAGCGCCAT	*

convert/convert.go
		answer = bed.Bed{Chrom: s.RName, Name: s.QName, FieldsInitialized: 4}
		if sam.IsPosStrand(s) {
			answer.ChromStart = int(s.Pos - 1) **// this was the 29-1=28 in the old test2.bed**
			answer.ChromEnd = numbers.Min(answer.ChromStart+fragLength-cigar.NumInsertions(s.Cigar)+cigar.NumDeletions(s.Cigar), reference[answer.Chrom].Size)
			answer.Strand = bed.Positive
		} else {
			answer.ChromEnd = int(s.Pos-1) + cigar.ReferenceLength(s.Cigar)
			answer.Strand = bed.Negative
			answer.ChromStart = numbers.Max(answer.ChromEnd-(fragLength-cigar.NumInsertions(s.Cigar)+cigar.NumDeletions(s.Cigar)), 0) **// this is the numbers.Max(33-(30-0+0),0) = 3 in the new test2.bed**
		}
		return answer

So maybe the new test2.bed is right?

Reply:
Yes, as Raven pointed out, they changed because both call the `convert.SamToBedFrag` function which is dependent on strand (specifically for handling reference length when the cigar contains indels).